interface field in literal composite struct

The following program `sample.go` triggers a panic:

```go
package main

import (
    "fmt"
    "io/ioutil"
    "log"
    "net/http"
    "net/http/httptest"
)

type mw1 struct {
    next http.Handler
}

func (m *mw1) ServeHTTP(rw http.ResponseWriter, rq *http.Request) {
    m.next.ServeHTTP(rw, rq)
}

type mw0 struct{}

func (m *mw0) ServeHTTP(w http.ResponseWriter, r *http.Request) {
    fmt.Fprint(w, "Welcome to my website!")
}

func main() {
    m0 := &mw0{}
    m1 := &mw1{next: m0}

    mux := http.NewServeMux()
    mux.HandleFunc("/", m1.ServeHTTP)

    server := httptest.NewServer(mux)
    defer server.Close()

    client(server.URL)
}

func client(uri string) {
    resp, err := http.Get(uri)
    if err != nil {
        log.Fatal(err)
    }
    body, err := ioutil.ReadAll(resp.Body)
    if err != nil {
        log.Fatal(err)
    }
    fmt.Println(string(body))
}
```

Expected result:
```console
$ go run ./sample.go
Welcome to my website!
```

Got:
```console
$ yaegi ./sample.go
/home/marc/yt/sample.go:26:9: panic
reflect.Set: value of type *struct {} is not assignable to type http.Handler
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000186c00, 0x40)
        /usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
github.com/containous/yaegi/interp.(*Interpreter).eval.func1(0xc0001e3c58)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:403 +0xc5
panic(0xdac280, 0xc0003e4300)
        /usr/lib/go/src/runtime/panic.go:969 +0x175
github.com/containous/yaegi/interp.runCfg.func1(0xc0001ce820, 0xc0003c6500)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:123 +0x217
panic(0xdac280, 0xc0003e4300)
        /usr/lib/go/src/runtime/panic.go:969 +0x175
reflect.Value.assignTo(0xd99e40, 0xc000136c30, 0x196, 0xee052d, 0xb, 0xe02420, 0xc0003e4130, 0x415145, 0xc0001e39d0, 0x4c51ec)
        /usr/lib/go/src/reflect/value.go:2425 +0x405
reflect.Value.Set(0xe02420, 0xc0003e4130, 0x194, 0xd99e40, 0xc000136c30, 0x196)
        /usr/lib/go/src/reflect/value.go:1554 +0xbd
github.com/containous/yaegi/interp.doCompositeSparse.func1(0xc0001ce820, 0xc0003e6190)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:2144 +0x16d
github.com/containous/yaegi/interp.runCfg(0xc0003c6500, 0xc0001ce820)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:129 +0x6e
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000370000, 0xc00037fc00, 0xc0001ce780)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:107 +0x2af
github.com/containous/yaegi/interp.(*Interpreter).eval(0xc000370000, 0xc0001b4e00, 0x30e, 0x7ffc03faf5a8, 0x15, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:495 +0x75c
github.com/containous/yaegi/interp.(*Interpreter).EvalPath(0xc000370000, 0x7ffc03faf5a8, 0x15, 0xc0001b4a00, 0x30e, 0x0, 0x0, 0x7ffc03faf5b6)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:387 +0xd5
main.runFile(0xc000370000, 0x7ffc03faf5a8, 0x15, 0xc000138d20, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/run.go:122 +0x17d
main.run(0xc000128030, 0x1, 0x1, 0xf1a358, 0xdbdb60)
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/run.go:80 +0x80e
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:128 +0x29b

```

This is a regression. It used to work for example in v0.8.12

--- a/_test/cli4.go
+++ b/_test/cli4.go
@@ -1,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+)
+
+type mw1 struct {
+	next http.Handler
+}
+
+func (m *mw1) ServeHTTP(rw http.ResponseWriter, rq *http.Request) {
+	m.next.ServeHTTP(rw, rq)
+}
+
+type mw0 struct{}
+
+func (m *mw0) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprint(w, "Welcome to my website!")
+}
+
+func main() {
+	m0 := &mw0{}
+	m1 := &mw1{m0}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", m1.ServeHTTP)
+
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	client(server.URL)
+}
+
+func client(uri string) {
+	resp, err := http.Get(uri)
+	if err != nil {
+		log.Fatal(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(string(body))
+}
+
+// Output:
+// Welcome to my website!

--- a/_test/cli5.go
+++ b/_test/cli5.go
@@ -1,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+)
+
+type mw1 struct {
+	next http.Handler
+}
+
+func (m *mw1) ServeHTTP(rw http.ResponseWriter, rq *http.Request) {
+	m.next.ServeHTTP(rw, rq)
+}
+
+type mw0 struct{}
+
+func (m *mw0) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprint(w, "Welcome to my website!")
+}
+
+func main() {
+	m0 := &mw0{}
+	m1 := &mw1{next: m0}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", m1.ServeHTTP)
+
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	client(server.URL)
+}
+
+func client(uri string) {
+	resp, err := http.Get(uri)
+	if err != nil {
+		log.Fatal(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(string(body))
+}
+
+// Output:
+// Welcome to my website!

--- a/interp/run.go
+++ b/interp/run.go
@@ -2078,9 +2078,16 @@ func doCompositeLit(n *node, hasType bool) {
 	values := make([]func(*frame) reflect.Value, len(child))
 	for i, c := range child {
 		convertLiteralValue(c, typ.field[i].typ.TypeOf())
-		if c.typ.cat == funcT {
+		switch {
+		case c.typ.cat == funcT:
 			values[i] = genFunctionWrapper(c)
-		} else {
+		case isArray(c.typ) && c.typ.val != nil && c.typ.val.cat == interfaceT:
+			values[i] = genValueInterfaceArray(c)
+		case isRecursiveType(typ.field[i].typ, typ.field[i].typ.rtype):
+			values[i] = genValueRecursiveInterface(c, typ.field[i].typ.rtype)
+		case isInterface(typ.field[i].typ):
+			values[i] = genInterfaceWrapper(c, typ.field[i].typ.rtype)
+		default:
 			values[i] = genValue(c)
 		}
 	}
@@ -2134,6 +2141,8 @@ func doCompositeSparse(n *node, hasType bool) {
 			values[field] = genValueInterfaceArray(c1)
 		case isRecursiveType(typ.field[field].typ, typ.field[field].typ.rtype):
 			values[field] = genValueRecursiveInterface(c1, typ.field[field].typ.rtype)
+		case isInterface(typ.field[field].typ):
+			values[field] = genInterfaceWrapper(c1, typ.field[field].typ.rtype)
 		default:
 			values[field] = genValue(c1)
 		}

